internal/feature_test.go

```diff
@@ -132,106 +142,134 @@ func TestScriptBasedCore(t *testing.T) {
 	}
 	defer feature.Disconnect(true)
 
-	testDownloadInstall(feature, mc, w.GenerateSoftwareArtifacts(false, "install"), t)
-
-	feature.serverCert = testCert
-	testDownloadInstall(feature, mc, wSecure.GenerateSoftwareArtifacts(true, "install"), t)
-}
-
-func testDownloadInstall(feature *ScriptBasedSoftwareUpdatable, mc *mockedClient, artifacts []*hawkbit.SoftwareArtifactAction, t *testing.T) {
-	// Preapare simple software update action.
-	sua := &hawkbit.SoftwareUpdateAction{
-		CorrelationID: "test-correlation-id",
-		SoftwareModules: []*hawkbit.SoftwareModuleAction{{
-			SoftwareModule: &hawkbit.SoftwareModuleID{Name: "test", Version: "1.0.0"},
-			Artifacts:      artifacts,
-			Metadata:       map[string]string{"artifact-type": "plane"},
-		}},
+	if basic {
+		testDownloadInstall(feature, mc, w.GenerateSoftwareArtifacts(false, "install"), t)
+		feature.serverCert = testCert
+		testDownloadInstall(feature, mc, wSecure.GenerateSoftwareArtifacts(true, "install"), t)
+	} else {
+		testResume(feature, mc, w.GenerateSoftwareArtifacts(false, "install"), t)
 	}
+}
 
-	// 2. Try to execute simple an download operation.
-	feature.downloadHandler(sua, feature.su)
+func testResume(feature *ScriptBasedSoftwareUpdatable, mc *mockedClient, artifacts []*hawkbit.SoftwareArtifactAction, t *testing.T) {
+	sua := prepareSoftwareUpdateAction(artifacts)
+	testReconnectWhileRunningOperation(feature, mc, sua, false, t) // disconnect while downloading
+	testReconnectWhileRunningOperation(feature, mc, sua, true, t)  // disconnect while downloading or installing
```

Comment:
As I can understand the different, `while downloading or installing` should be: `while installing`

Reply:
It is not sure when disconnect will happen(while download or install), since it is executed in a separate thread. The goal is to have a completed operation, after reconnect.